extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java

```diff
@@ -657,14 +648,81 @@ public String apply(String className) {
                             }
                         }));
                 generatedTypes.put(localizedFile.toString(),
-                        generateImplementation(bundle.getDefaultBundleInterface(), bundleImpl, bundleInterface,
-                                localeAwareGizmoAdaptor,
-                                messageTemplateMethods, keyToTemplate, locale));
+                        generateImplementation(bundleInterface, bundleImpl, new SimpleClassInfoWrapper(bundleInterface),
+                                localeAwareGizmoAdaptor, messageTemplateMethods, keyToTemplate, locale));
             }
         }
         return generatedTypes;
     }
 
+    private Map<String, String> getLocalizedFileKeyToTemplate(MessageBundleBuildItem bundle,
+            ClassInfo bundleInterface, String locale, List<MethodInfo> methods, ClassInfo localizedInterface)
+            throws IOException {
+
+        var localizedFile = bundle.getMergeCandidates().get(locale);
+        if (nonNull(localizedFile)) {
```

Comment:
What's the point of `nonNull()` here? I mean what's wrong with `localizedFile != null`? ;-)

Reply:
Not really, it reads better to me, but it's irrelevant, I switched to != null and == null